src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -257,6 +259,7 @@ private void ClearCompilationScopeMap(AnalyzerAndOptions analyzerAndOptions, Ana
             Tuple<ImmutableArray<DiagnosticDescriptor>, EventHandler<Exception>> value;
             lock (_descriptorMap)
             {
+                // REVIEW: how one knows exception handler is same as the one cached?
```

Comment:
cache cached descriptor, but descriptor can have different exception handler subscribed to it. so if one uses cached descriptor without checking whether given exception handler is subscribed to the descriptor, returned value could be wrong. @mavasani should be able to explain more detail.